### ISSUE TITLE
Enable Link-Time Optimization (LTO)

### ISSUE THREAD

**Issue by zamazan4ik** [phiresky/ripgrep-all#254]
**Is your feature request related to a problem? Please describe.**
Not a problem. Just an improvement idea for binary size and, possibly, performance

**Describe the solution you'd like**
I noticed that in the `Cargo.toml` file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit (not the case for the application but anyway - we are blazingly fast here).

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where in addition to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. Using ThinLTO should also help to reduce the build-time overhead with LTO. E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48).

Basically, it can be enabled with the following lines:
```
[profile.release]
debug = true
split-debuginfo = "packed"
lto = true
```

I made quick tests locally (latest commit from the master branch, Fedora 40) - enabling LTO decreases the binary size in Release mode from 46 Mib to 27 Mib for the `rga` binary.

**Describe alternatives you've considered**
Leave the things as is :)

**Additional context**
N/A

**Comment by lafrenierejm** [comment]
Is there a particular reason to set `debug = true` for release builds? I would have thought it would make more sense to leave debug symbols disabled…

**Comment by zamazan4ik** [comment]
I don't know, tbh - it's already written in the Cargo.toml file - I just added `lto = true` to the Release profile
